plugin/path_login.go

```diff
@@ -141,6 +156,22 @@ type gcpLoginInfo struct {
 	GceMetadata *gcputil.GCEIdentityMetadata
 }
 
+func (b *GcpAuthBackend) parseRoleNameAndRole(ctx context.Context, s logical.Storage, data *framework.FieldData) (string, *gcpRole, error) {
+	roleName := data.Get("role").(string)
+	if roleName == "" {
+		return "", nil, errors.New("role is required")
+	}
+
+	role, err := b.role(ctx, s, roleName)
+	if err != nil {
+		return "", nil, err
+	}
+	if role == nil {
+		return "", nil, fmt.Errorf("role '%s' not found", roleName)
```

Comment:
nit: You can make use of `%q` instead of `'%s'` here.

Reply:
I think it's better to update and improve instead of keeping bad patterns, please do feel free to change this to `%q`